test/catch2/ext/test_key_provider.cpp

```diff
@@ -0,0 +1,265 @@
+    {
+        kp_ptr_t kp = setup(nullptr);
+        REQUIRE(kp->wtext != nullptr);
+
+        REQUIRE(kp->verbose == WT_VERBOSE_INFO);
+        REQUIRE(kp->key_expires == 0);
+    }
+
+    SECTION("Empty config")
+    {
+        kp_ptr_t kp = setup("");
+        REQUIRE(kp->wtext != nullptr);
+
+        REQUIRE(kp->verbose == WT_VERBOSE_INFO);
+        REQUIRE(kp->key_expires == 0);
+    }
+
+    SECTION("Custom config")
+    {
+        kp_ptr_t kp = setup("verbose=2,key_expires=300");
+        REQUIRE(kp->wtext != nullptr);
+
+        REQUIRE(kp->verbose == WT_VERBOSE_DEBUG_2);
+        REQUIRE(kp->key_expires == 300);
+        REQUIRE(kp->state.current_lsn == 0);
+        REQUIRE(kp->state.key_time == 0);
+        REQUIRE(kp->state.current_key == nullptr);
+        REQUIRE(kp->state.key_size == 0);
+    }
+
+    SECTION("Invalid config")
+    {
+        const char *invalid_configs[] = {
+          "verbose=invalid",
+          "key_expires=invalid",
+          "verb=0,key=1",
+          "wrong=bad",
+        };
+
+        for (const char *config : invalid_configs) {
+            const char *ext_config[] = {config, nullptr};
+            int ret = extension_init(conn, reinterpret_cast<WT_CONFIG_ARG *>(ext_config));
+            REQUIRE(ret == EINVAL);
+        }
+    }
+}
+
+TEST_CASE_METHOD(kp_fixture, "Key never expires", "[key_provider]")
+{
+    /* By default, the key never expires. */
+    kp_ptr_t kp = setup(nullptr);
+    REQUIRE(kp->wtext != nullptr);
+
+    WT_KEY_PROVIDER *wtkp = &kp->iface;
+
+    /* Initial state */
+    REQUIRE(kp->verbose == 0);
+    REQUIRE(kp->key_expires == 0);
+    REQUIRE(kp->state.current_lsn == 0);
+    REQUIRE(kp->state.key_time == 0);
+    REQUIRE(kp->state.current_key == nullptr);
+    REQUIRE(kp->state.key_size == 0);
+
+    /* Dummy key and LSN */
+    const std::string dummy_key = "dummy_key_data";
+    const uint64_t dummy_lsn = 42;
+
+    /* Load key */
+    WT_CRYPT_KEYS crypt_in = {};
+    crypt_in.r.lsn = dummy_lsn;
+    crypt_in.keys.data = dummy_key.data();
+    crypt_in.keys.size = dummy_key.size();
+
+    REQUIRE(wtkp->load_key(wtkp, session(), &crypt_in) == 0);
+
+    REQUIRE(kp->state.current_lsn == dummy_lsn);
+    REQUIRE(memcmp(kp->state.current_key, dummy_key.data(), dummy_key.size()) == 0);
+    REQUIRE(kp->state.key_size == dummy_key.size());
+    REQUIRE(kp->state.key_time != 0);
+
+    /* Probe the key; the key never expires */
+    WT_CRYPT_KEYS crypt_out = {};
+    REQUIRE(wtkp->get_key(wtkp, session(), &crypt_out) == 0);
+    REQUIRE(crypt_out.r.lsn == 0);
+    REQUIRE(crypt_out.keys.size == 0); /* Key has not changed */
+    REQUIRE(crypt_out.keys.data == nullptr);
+}
+
+TEST_CASE_METHOD(kp_fixture, "Key expire", "[key_provider]")
+{
+    /* Key expiration period = 12 hours */
+    kp_ptr_t kp = setup("verbose=1,key_expires=43200");
+    REQUIRE(kp->wtext != nullptr);
+
+    WT_KEY_PROVIDER *wtkp = &kp->iface;
+
+    REQUIRE(kp->verbose == WT_VERBOSE_DEBUG_1);
```

Comment:
Can we provide a helper function that performs this checking for us?

Reply:
Not sure what you mean. The test case should assert the state of the object.

Reply:
Talked offline, and we are sticking with this